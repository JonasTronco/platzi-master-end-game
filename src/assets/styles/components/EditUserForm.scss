@import '../main.scss';

.avatar-form {
  @include flexStaCen;
  margin-bottom: $space_x3;

  &__preview {
    width: 90px;
    height: 90px;
    margin-right: $space_x3;
    padding: 0;
    border-radius: 50%;
    box-shadow: $shadow;
    overflow: hidden;

    &__img {
      width: 100%;
      height: 100%;
    }
  }
}

.edit-user-form {
  @include colStaSta;

  &__title {
    margin-top: $space_x6;
  }

  &__section {
    @include flexStaSta;
    flex-wrap: wrap;

    .input {
      width: 300px;
      margin: $space_x3 $space_x6 0 0;

      &__label,
      &__field,
      &__error {
        width: 100%;
      }
    }
  }

  button {
    align-self: flex-end;
    margin: $space_x6 $space_x6 0 0;
  }

  &__load {
    align-self: flex-end;
    margin-right: $space_x6;
  }
}
