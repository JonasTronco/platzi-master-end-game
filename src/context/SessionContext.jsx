import { useState, useEffect, createContext, useMemo, useContext } from 'react';
import Router from 'next/router';

const SessionContext = createContext();
export function SessionProvider(props) {
  const [session, setSession] = useState(null);

  useEffect(() => {
    try {
      setSession(JSON.parse(localStorage.getItem('session')));
    } catch (error) {
      Router.push('/');
    }
  }, []);

  const value = useMemo(
    () => ({
      session,
    }),
    [session],
  );

  return <SessionContext.Provider value={value} {...props} />;
}

export function useSession() {
  const context = useContext(SessionContext);
  if (!context) {
    throw new Error('No Session Context Provider');
  }
  return context;
}
