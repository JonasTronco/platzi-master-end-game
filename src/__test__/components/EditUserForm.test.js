import EditUserForm from '../../components/EditUserForm';
import { create } from 'react-test-renderer';
import { render, fireEvent, act, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Router from 'next/router';
import { shallow, mount } from 'enzyme';
import { SessionProvider } from '../../context/SessionContext';
describe('EditUserForm Component', () => {
  const mockedRouter = { push: () => {} };
  Router.router = mockedRouter;
  afterEach(cleanup);
  const obj = {
    session: {
      token:
        'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiI1ZWM1Y2UxNmZjMTNhZTE1MDYwMDAwNjQiLCJ1c2VybmFtZSI6Im1heW5lLnNuYXNlbC40MTcxIiwic2NvcGVzIjpbXSwiaWF0IjoxNTkwNzM4OTc3LCJleHAiOjE1OTExNzA5Nzd9.5gPpAnLLjKD4-O8oWsbZ860Iy0JBNjW175QxV0PCur4',
      user: {
        id: '5ec5ce16fc13ae1506000064',
        username: 'mayne.snasel.4171',
        typeOfUser: 'Patient',
        isActive: false,
        imageURL: 'http://dummyimage.com/200x200.png/ff4444/ffffff',
        firstName: 'Mayne',
        lastName: 'Snasel',
        defaultPath: '/tests'
      }
    }
  };
  test('Should render EditUserForm component', () => {
    const component = create(
      <SessionProvider value={obj}>
        <EditUserForm setUpdateLoading={true} userID='freddy.van.1009' />
      </SessionProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  // test('Hola test edit ', () => {
  //   const onSubmit = jest.fn();
  //   const component = render(
  //     <SessionProvider value={obj}>
  //       <EditUserForm userID='' />
  //     </SessionProvider>
  //   );
  //   act(() => {
  //     fireEvent.click(component.getByTestId('button-submited'));
  //   });
  //   expect(onSubmit).toHaveBeenCalled();
  // });
  // test('Should rendered of EditUserForm component', () => {
  //   const edit = render(
  //     <SessionProvider value={obj}>
  //       <EditUserForm userID='freddy.van.1009' />
  //     </SessionProvider>
  //   );

  //   expect(edit.getByTestId('useredit')).toBeInTheDocument();
  // });
  describe('<EditUserForm />', () => {
    afterEach(cleanup);
    test('Render del componente Employee', () => {
      const editUser = mount(
        <SessionProvider value={obj}>
          <EditUserForm setUpdateLoading={true} userID='freddy.van.1009' />
        </SessionProvider>
      );
      expect(editUser.length).toEqual(1);
    });
    test('Render del componente Employee', () => {
      const editUser = shallow(
        <SessionProvider value={obj}>
          <EditUserForm setUpdateLoading={true} userID='freddy.van.1009' />
        </SessionProvider>
      );
      expect(editUser.length).toEqual(1);
    });
  });
});
