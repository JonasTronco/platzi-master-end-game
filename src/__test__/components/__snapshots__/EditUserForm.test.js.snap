// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditUserForm Component Should render EditUserForm component 1`] = `
Array [
  "",
  "",
]
`;
